### ISSUE TITLE
GLX: Failed to find a suitable GLXFBConfig

### ISSUE THREAD

**Issue by Y2Kwastaken** [PrismLauncher/PrismLauncher#866]
### Operating System

Linux

### Version of Prism Launcher

Prism Launcher 6.3

### Version of Qt

Qt 5.15.8

### Description of bug

Expected functionality would be the client starting

What happened:

```
[15:55:24] [Render thread/INFO]: Backend library: LWJGL version 3.2.2 build 10
Gtk-Message: 15:55:24.522: Failed to load module "canberra-gtk-module"
Gtk-Message: 15:55:24.523: Failed to load module "canberra-gtk-module"
sh: line 3: xprop: command not found
sh: line 3: xprop: command not found
[15:55:24] [Render thread/WARN]: Failed to create window: 
dsr$a: GLFW error 65545: GLX: Failed to find a suitable GLXFBConfig
	at dsr.b(SourceFile:218) ~[minecraft-1.18.2-client.jar:?]
	at org.lwjgl.glfw.GLFWErrorCallbackI.callback(GLFWErrorCallbackI.java:36) ~[lwjgl-glfw-3.2.2.jar:build 10]
	at org.lwjgl.system.JNI.invokePPPP(Native Method) ~[lwjgl-3.2.2.jar:build 10]
	at org.lwjgl.glfw.GLFW.nglfwCreateWindow(GLFW.java:1714) ~[lwjgl-glfw-3.2.2.jar:build 10]
	at org.lwjgl.glfw.GLFW.glfwCreateWindow(GLFW.java:1897) ~[lwjgl-glfw-3.2.2.jar:build 10]
	at dsr.<init>(SourceFile:92) ~[minecraft-1.18.2-client.jar:?]
	at erh.a(SourceFile:21) ~[minecraft-1.18.2-client.jar:?]
	at dyr.<init>(SourceFile:476) ~[minecraft-1.18.2-client.jar:?]
	at net.minecraft.client.main.Main.main(SourceFile:197) [minecraft-1.18.2-client.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71) [NewLaunch.jar:?]

```

### Steps to reproduce

* Create a new instance
* Start the Instance
* Error Appears

### Suspected cause

It may have to do with the version of NVIDIA graphics drivers or LWJGL version I use. 

Another note I had this issue with neither MultiMC nor FTB Launcher which I switched to when this issue started.

### This issue is unique

- [ ] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by Y2Kwastaken** [comment]
System NVIDIA drivers and Flatpak NVIDIA drivers were not synced in version

In my case
**Flatpak**: ```NVIDIA drivers version 515```
**System**: ```NVIDIA drivers version 525```

**Soltuion**
- Close Prism Launcher
- Update Drivers `flatpak update`
- Reopen Prism Launcher and continue!

**Comment by dennissherb** [comment]
Could you explain how exactly you fixed that? how did you determine that the versions were mismatched?

**Comment by ws909** [comment]
On Pop!_OS. 
Installed these updates from Cosmic Store, and that borked MC launched from Prism. The vanilla launcher launches the game just fine.
```
aug. 24 13:06:37 pull           org.freedesktop.Platform.GL.nvidia-550-67      1.4    user         flathub
aug. 24 13:06:39 deploy update  org.freedesktop.Platform.GL.nvidia-550-67      1.4    user         flathub
aug. 24 13:07:28 pull           org.freedesktop.Platform.GL.nvidia-555-58-02   1.4    user         flathub
aug. 24 13:07:30 deploy update  org.freedesktop.Platform.GL.nvidia-555-58-02   1.4    user         flathub
aug. 24 13:08:19 pull           org.freedesktop.Platform.GL32.nvidia-555-58-02 1.4    user         flathub
aug. 24 13:08:20 deploy update  org.freedesktop.Platform.GL32.nvidia-555-58-02 1.4    user         flathub
```

I tried [the suggested solution](https://github.com/PrismLauncher/PrismLauncher/issues/866#issuecomment-1432251741). It does not work. Reintalling Prism neither works:
```
flatpak uninstall org.prismlauncher.PrismLauncher
flatpak install org.prismlauncher.PrismLauncher
```

The exact error: net.minecraft.class_1041$class_4716: GLFW error 65545: GLX: Failed to find a suitable GLXFBConfig

**Comment by BenjaminBrienen** [comment]
I get this now @Y2Kwastaken 

```
[23:56:50] [Render thread/INFO]: Modifying process environment to apply workarounds for the NVIDIA graphics driver...
[23:56:53] [Render thread/WARN]: Failed to create window: 
net.minecraft.class_1041$class_4716: GLFW error 65545: GLX: Failed to find a suitable GLXFBConfig
	at knot/net.minecraft.class_1041.method_4501(class_1041.java:217) ~[client-intermediary.jar:?]
	at knot/org.lwjgl.glfw.GLFWErrorCallbackI.callback(GLFWErrorCallbackI.java:43) ~[lwjgl-glfw-3.3.3.jar:?]
	at knot/org.lwjgl.system.JNI.invokePPPP(Native Method) ~[lwjgl-3.3.3.jar:?]
	at knot/org.lwjgl.glfw.GLFW.nglfwCreateWindow(GLFW.java:2058) ~[lwjgl-glfw-3.3.3.jar:?]
	at knot/org.lwjgl.glfw.GLFW.glfwCreateWindow(GLFW.java:2229) ~[lwjgl-glfw-3.3.3.jar:?]
	at knot/net.minecraft.class_1041.redirect$cme000$sodium$wrapGlfwCreateWindow(class_1041.java:1112) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_1041.mixinextras$bridge$redirect$cme000$sodium$wrapGlfwCreateWindow$58(class_1041.java) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_1041.wrapOperation$cjf000$sodium$setAdditionalWindowHints(class_1041.java:603) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_1041.<init>(class_1041.java:108) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_3682.method_16038(class_3682.java:21) ~[client-intermediary.jar:?]
	at knot/net.minecraft.class_310.<init>(class_310.java:512) ~[client-intermediary.jar:?]
	at knot/net.minecraft.client.main.Main.main(Main.java:234) [client-intermediary.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:506) [fabric-loader-0.17.2.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:72) [fabric-loader-0.17.2.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.17.2.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:105) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:129) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?]
Proces is afgesloten met code 0.

```

**Comment by Petros626** [comment]
@BenjaminBrienen what have you exactly done to end there?

**Comment by BenjaminBrienen** [comment]
I don't know. Maybe it is related to my monitor being 8k, 10 bit. Sometimes I will change a mod and the error happens, but it randomly goes away.

**Comment by Petros626** [comment]
What about installing fresh GPU drivers on your system? I think in my case a upgraded gpu driver for a newer cuda version destroyed my the GLX.

**Comment by BenjaminBrienen** [comment]
I am on the latest drivers available, nvidia 580.82.09.

**Comment by Parciwal** [comment]
If you still have this problem: it fixed it for me when I started minecraft once through the official launcher.

**Comment by m1cr0man** [comment]
For me, this occurred on NixOS because I updated my system and did not update the version of Java that was set in my instance configuration. It was trying to use an older openJDK path in the nix store than the newer one I had available (17.0.16 vs 17.0.18).

For others encountering this I would suggest trying to "Detect" the Java executable and selecting the right one again.

**Comment by stichiboi** [comment]
> Could you explain how exactly you fixed that? how did you determine that the versions were mismatched?

In the Konsole you can run 
```
flatpak list

nvidia-smi
```

And you should look for something like:
```
nvidia-580-142                             org.freedesktop.Platform.GL.nvidia-580-142                                          1.4             system
```
and
```
NVIDIA-SMI 580.173.02             Driver Version: 580.173.02     CUDA Version: 13.0   
....
